cmd/cleanupexamples/main.go

```diff
@@ -81,6 +81,10 @@ func processYAML(yamlData []byte) ([]byte, error) {
 			return nil, fmt.Errorf("cannot decode the YAML file: %w", err)
 		}
 
+		if u == nil {
```

Comment:
Maybe we should do the nil check before the `decoder.Decode(&u);` call, since I assume there's no value in doing that if its nil?

Reply:
Or does the decoded.Decode actually cause `u` to become nil?

Reply:
Nevermind I looked up what the `&` does in Go.

Reply:
The value of `u` is the “output” of `decoder.Decode(&u)`, so doing the nil check before is not possible.

Reply:
You already found out while I was writing 🙂

Reply:
Yes, my first assumption was in that direction, but here, the value of the `u` becomes nil after the decoder process.